### ISSUE TITLE
Bad data showing up in summary table and deployment plot

### ISSUE THREAD

**Issue by jsayers-bsc** [MotusDev/Motus-TO-DO#878]
Receiver deployment 7338.

[daily summary](https://motus.org/data/receiverDeployment?id=7338)
[deployment plot](https://motus.org/data/receiver/deployment/plot/?id=7338)

The only tag that shows up in the daily summary is a Lazuli Bunting — which raises an eyebrow or two. The tag also shows up in the deployment plot, along with several other tags. 

However, when I download and summarize the "good" tags at this station, this tag is not among them. In fact, there is _only one single hit_ from this tag in the data. Additionally, none of the other tags in the deployment plot are represented by more than 1 hit; everything in the deployment plot is runLen == 1.

There are, however, 22 tags which _do_ have run lengths of more than 3. They are all CTT tags and include many familiar frequent fliers such as TRES 61330000 (unclear at a glance if any of them are legit). But none of these show up in the daily summary or the deployment plot, despite many having run lengths in the thousands. (Some are bogus and shouldn't show up there anyway, but I didn't think we'd yet addressed these tags).

**Comment by cjardine-bsc** [comment]
I started to investigate. 

The problem is not with run filters, those are not run for CTT tags. The below returns 0 records. 

SELECT TOP (1000) * 
  FROM [motus].[dbo].[det_runs_filters] a inner join tags b on a.tagID =b.id 
  AND manufacturer = 'CTT'
  
  
RE: However, when I download and summarize the "good" tags at this station, this tag is not among them. In fact, there is only one single hit from this tag in the data. Additionally, none of the other tags in the deployment plot are represented by more than 1 hit; everything in the deployment plot is runLen == 1.
 
That's right, we don't filter on run length for CTT tags at the moment. So a run length of 1 from a CTT tag would still be considered "good" data. (for now!)

RE: There are, however, 22 tags which do have run lengths of more than 3. They are all CTT tags and include many familiar frequent fliers such as TRES 61330000 (unclear at a glance if any of them are legit). But none of these show up in the daily summary or the deployment plot, despite many having run lengths in the thousands. (Some are bogus and shouldn't show up there anyway, but I didn't think we'd yet addressed these tags).

Stu ended a bunch of deployments for those problematic TRES tags, which likely explains why they are being excluded from the public summaries, despite their over zealous run lengths. No deployment, no display on public pages. 

As for the 12 tags on the deployment plot: https://motus.org/data/receiver/deployment/plot/?id=7338

A spot check shows that most of them are not registered to a project yet (currently under project 0) or they don't have deployment metadata (no deployment mo' problems). 

SELECT  * FROM [motus].[dbo].[tags] a left join tags_deployments b on a.id=b.tag_id 
where a.id in (49587, 42890, 46426, 49579,49196)
  
So....I know you're not going to like it, but I think everything is working as expected here. 
  
Better filtering for CTT tags is on the to-do list.

**Comment by denislepage** [comment]
I suspect we should put some basic filter in place to exclude very short CTT runs. The solar tags are normally emitting signals much more frequently than battery tags (~1 per s), so single hit runs should be much more rarely a valid fly-by. Not sure what the burst intervals is for power tags, but probably the same.

Until we have a better handle on this, I think we should exclude at the very least runs of 1 (single hits) from the public outputs to start.

False positives involving runs longer than 1 are a different kettle of fish. If they were truly random, we should virtually never see them (2 consecutive hits being the same ID purely by chance is 1/ 18,446,744,073,709,551,616). However, there are some common patterns such as trailing zeros that will increase their probability significantly, as stated, and we probably need a different approach for those. I am pretty sure that excluding runs of 2 would not solve these types of problems, though it could help, perhaps, while running the risk of filtering out more good detections than bad ones.  Changes made on CTT’s end recently should address several of those cases, and we could exclude the more problematic TRES tags entirely if not already done.

**Comment by cjardine-bsc** [comment]
Okie. I am continuing the cleaning momentum! 

I've added a process to prSGBatchesActivityBatchID that will insert runs with a length of 1 for ctt batches into det_runs_filters.  

This will need to be run for all the existing CTT batches. They will need their det_taghits_daily summaries re-run too. I've inserted them into the procs queue in live, it will take a bit of time to crunch through those (there are 21k CTT batches with runs of  len =1) but I'll keep an eye on it this evening.

All new batches will have this filter applied upon import.
